core/primitives/piece/impl/piece_data.cpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "primitives/piece/piece_data.hpp"
+
+#include <fcntl.h>
+#include <boost/filesystem.hpp>
+
+namespace fc::primitives::piece {
+  PieceData::PieceData(const std::string &path_to_file) {
+    fd_ = -1;
```

Comment:
```suggestion
  PieceData::PieceData(const std::string &path_to_file) : fd_{-1} {
```

Maybe use named constant for -1, also add bool query if opened or -1

Reply:
Fixed